packages/language-server/src/__test__/completion.test.ts

```diff
@@ -1190,6 +1252,303 @@ suite('Completions', function () {
+            @@index([address.a|])
+          }`,
+        expected: {
+          isIncomplete: false,
+          items: [],
+        },
+      })
+    })
+    test('MongoDB: @@index([email, address.|]) with composite type suggestion, depth 1', () => {
+      assertCompletion({
+        provider: 'mongodb',
+        schema: /* Prisma */ `
+          type Address {
+            street String
+            number Int
+            alpha  Alpha
+          }
+          type Alpha {
+            bravo  Bravo
+            helloA Int
+          }
+          type Bravo {
+            something  String
+            helloBravo Int
+          }
+          model User {
+            id      Int     @id
+            email   String
+            address Address
+            @@index([email, address.|])
+          }`,
+        expected: {
+          isIncomplete: false,
+          items: [
+            { label: 'street', kind: CompletionItemKind.Field },
+            { label: 'number', kind: CompletionItemKind.Field },
+            { label: 'alpha', kind: CompletionItemKind.Field },
+          ],
+        },
+      })
+    })
+    test('MongoDB: @@index([email, address.alpha.|]) with composite type suggestion, depth 2', () => {
+      assertCompletion({
+        provider: 'mongodb',
+        schema: /* Prisma */ `
+          type Address {
+            street String
+            number Int
+            alpha  Alpha
+          }
+          type Alpha {
+            bravo  Bravo
+            helloA Int
+          }
+          type Bravo {
+            something  String
+            helloBravo Int
+          }
+          model User {
+            id      Int     @id
+            email   String
+            address Address
+            @@index([email, address.alpha.|])
+          }`,
+        expected: {
+          isIncomplete: false,
+          items: [
+            { label: 'bravo', kind: CompletionItemKind.Field },
+            { label: 'helloA', kind: CompletionItemKind.Field },
+          ],
+        },
+      })
+    })
+    test('MongoDB: @@index([email, address.alpha.bravo.|]) with composite type suggestion, depth 3', () => {
+      assertCompletion({
+        provider: 'mongodb',
+        schema: /* Prisma */ `
+          type Address {
+            street String
+            number Int
+            alpha  Alpha
+          }
+          type Alpha {
+            bravo  Bravo
+            helloA Int
+          }
+          type Bravo {
+            something  String
+            helloBravo Int
+          }
+          model User {
+            id      Int     @id
+            email   String
+            address Address
+            @@index([email, address.alpha.bravo.|])
+          }`,
+        expected: {
+          isIncomplete: false,
+          items: [
+            { label: 'something', kind: CompletionItemKind.Field },
+            { label: 'helloBravo', kind: CompletionItemKind.Field },
+          ],
+        },
+      })
+    })
```

Comment:
Here we might be missing the tests multiple levels deep in the composite types where we already have part of the completion types out, so e.g. ` @@index([email, address.alpha.bravo.hello|]` and it should only suggest `helloBravo`.

Reply:
Added test with comment 
```
// TODO, see if we can have better suggestions here, should suggest `helloBravo`
```